libs/blocks/table-of-contents/table-of-contents.js

```diff
@@ -0,0 +1,86 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+ * Table of Contents
+ */
+function findAnchorTarget(text) {
+  let linkText = text.toLowerCase();
+  linkText = linkText.charAt(0) === '_' ? linkText.substring(1) : linkText;
+  linkText = linkText.replaceAll('_', '-');
+  linkText = linkText.replaceAll(/[ /|&;$%@"<>()+,.]/g, '');
+  return document.querySelector(`[id^="${linkText}"]`);
+}
+
+const DOWN_ARROW_ICON = `<svg xmlns="http://www.w3.org/2000/svg" width="10.875" height="13.323" viewBox="0 0 10.875 13.323">
+  <g id="Group_177418" data-name="Group 177418" transform="translate(223.353 -1276.435) rotate(90)">
+    <line id="line_color_border_" data-name="line [color_border]" x2="10.909" transform="translate(1277.435 217.875)" fill="none" stroke="#1473e6" stroke-linecap="round" stroke-width="2"/>
+    <g id="arrow_head" data-name="arrow head" transform="translate(1284.32 213.892) rotate(45)">
+      <line id="line_color_border_2" data-name="line [color_border]" x2="5.69" transform="translate(0 0)" fill="none" stroke="#1473e6" stroke-linecap="round" stroke-width="2"/>
+      <line id="line_color_border_3" data-name="line [color_border]" y2="5.69" transform="translate(5.69 0)" fill="none" stroke="#1473e6" stroke-linecap="round" stroke-width="2"/>
+    </g>
+  </g>
+</svg>`;
+
+function getItemHTML(title, description, target) {
+  return `
+  <li class="toc-item">
+    <div class="toc-link-text">
+      ${target ? `<a class="section-title" href="#${target?.id}" target="_self">${title}</a>` : `<p class="section-title">${title}</p>`}
+      ${description ? `<p class="section-description">${description}</p>` : ''}
+    </div>
+    <div class="toc-arrow">
+      ${DOWN_ARROW_ICON}
+    </div>
+  </li>`;
+}
+
+export default function init(el) {
+  const children = Array.from(el.querySelectorAll(':scope > div'));
+  const title = children.shift().textContent;
+  const pageTop = document.querySelector('header')?.offsetHeight ?? 0;
+
+  const list = children.reduce((prev, section) => {
+    const sectionTitle = section.querySelector('strong');
+    const link = section.querySelector('a');
+    const subtitle = section.querySelectorAll('p').length > 2 ? section.querySelectorAll('p')[1] : null;
+    const target = findAnchorTarget(link.textContent);
+    const item = getItemHTML(sectionTitle?.textContent, subtitle?.textContent, target);
+
+    return prev + item;
+  }, '');
+  const sectionNav = `
+  <section class="container toc-container">
+    <p class="toc-title">${title}</p>
+    <nav aria-label="Table of contents">
+      <ul class="toc-list">${list}</ul>
+    </nav>
+  </section>`;
+
+  el.innerHTML = sectionNav;
+  el.querySelectorAll('.toc-item').forEach((item) => {
```

Comment:
It's generally a bad practice to use innterhtml and then query for items you just created.

I would highly recommend using document fragments + createTag throughout this document.

Reply:
This has been updated to use createTag and queries have been removed.